File: frontend/lib/types/employee.ts
export interface Employee {
  id: number;
  employee_code: string;
  name: string;
  email: string | null;
  phone: string | null;
  designation: string | null;
  department: string | null;
  join_date: string | null;
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

export interface EmployeeCreate {
  employee_code: string;
  name: string;
  email?: string | null;
  phone?: string | null;
  designation?: string | null;
  department?: string | null;
  join_date?: string | null;
  is_active?: boolean;
}

export interface EmployeeUpdate {
  name?: string;
  email?: string | null;
  phone?: string | null;
  designation?: string | null;
  department?: string | null;
  join_date?: string | null;
  is_active?: boolean;
}

export interface EmployeeListParams {
  q?: string;
  is_active?: boolean;
  department?: string;
  limit?: number;
  offset?: number;
}

File: backend/app/schemas/employee.py
from __future__ import annotations

from datetime import date, datetime

from pydantic import BaseModel, EmailStr, Field

from app.schemas.common import ORMModel


class EmployeeBase(BaseModel):
    employee_code: str = Field(min_length=1, max_length=64)
    name: str = Field(min_length=1, max_length=128)
    email: EmailStr | None = None
    phone: str | None = Field(default=None, max_length=32)
    designation: str | None = Field(default=None, max_length=128)
    department: str | None = Field(default=None, max_length=128)
    join_date: date | None = None


class EmployeeCreate(EmployeeBase):
    is_active: bool = True


class EmployeeUpdate(BaseModel):
    name: str | None = Field(default=None, min_length=1, max_length=128)
    email: EmailStr | None = None
    phone: str | None = Field(default=None, max_length=32)
    designation: str | None = Field(default=None, max_length=128)
    department: str | None = Field(default=None, max_length=128)
    join_date: date | None = None
    is_active: bool | None = None


class EmployeeRead(ORMModel):
    id: int
    employee_code: str
    name: str
    email: str | None
    phone: str | None
    designation: str | None
    department: str | None
    join_date: date | None
    is_active: bool
    created_at: datetime
    updated_at: datetime
